### ISSUE TITLE
[2019/06/26] Machine Learning 輪講

### ISSUE THREAD

**Issue by agatan** [wantedly/machine-learning-round-table#7]
## Why

Machine Learning 輪講は最新の技術や論文を追うことで、エンジニアが「技術で解決できること」のレベルをあげていくことを目的にした会です。

prev. #6 

## What

話したいことがある人はここにコメントしましょう！
面白いものを見つけた時点でとりあえず話すという宣言だけでもしましょう！

**Comment by MizuTakeuchi** [comment]
FYI
Pydata.Tokyoにご参加された町岡さんが19時より参加希望です。
@agatan

**Comment by Hayashi-Yudai** [comment]
## ArcFace: Additive Angular Margin Loss for Deep Face Recognition
https://arxiv.org/pdf/1801.07698.pdf
* 顔認証で各クラス同士の間のマージンが大きくなるように学習するための損失関数の提案

### 従来の方法
* softmax classifier
  * 分類するクラス数に線形にtransformation matrixが巨大化
  * 汎化性能が不十分
* triplet-loss
  * 3つ組の数が巨大なデータセットでは爆発的に増大
  * semi-hard sampleを見つけるのが難しい
* Sphereface
  * angular marginを導入したが、lossを計算するのに近似が必要でtrainingが不安定になりがち

### ArcFace
backboneから出てきた出力xと、重みパラメータWをそれぞれL2ノルムで正規化
-> この２つの行列積を取ると、得られたベクトルの各成分はxとWのそれぞれの行のcosになる。
-> 正解ラベルの成分だけ角度をmだけ大きくするペナルティを加えることでより明確にクラス分けを実現する

![image](https://user-images.githubusercontent.com/34836226/60086507-225d8480-9776-11e9-928a-742f4cdbca59.png)

**Comment by agatan** [comment]
## Do Neural Dialog Systems Use the Conversation History Effectively? An Empirical Study

ACL 2019 の short paper で、対話応答システムがちゃんと対話履歴の情報を使えているのか調べた論文。
対話履歴の情報にノイズをのせ、それによるデグレの量を調べている。（大きく落ちるなら対話履歴をちゃんと理解している、落ちないなら対話履歴にかかわらずそれっぽい応答を生成しているだけ）

ノイズは「対話履歴の順番をシャッフル / 逆順にする」「一文内の単語順をいじる」「特定の品詞の単語をマスクする」など。

### 結果


![image](https://user-images.githubusercontent.com/6925889/60148158-ecb6ab00-980a-11e9-8a72-624378ab41e1.png)


- Transformer ベースのモデルは順番の変化に鈍感
- Attention を使うモデルは、 `Only last` のときに大きくパフォーマンスが変化する = vanilla seq2seq と比べて対話初期のコンテキストを参照できている

**Comment by agatan** [comment]
## [WIP] Transformer-XL: Attentive Language Models Beyond a Fixed-Length Context

BERT を超えるモデルが出てきた [[1906.08237] XLNet: Generalized Autoregressive Pretraining for Language Understanding](https://arxiv.org/abs/1906.08237)
XLNet は Transformer-XL ベースらしいので、Transformer-XL のほうから読んでみる。
（紛らわしいことに facebookresearch も BERT を超えた XLM というのを出していた。こっちは cross lingual の XL かな？ https://arxiv.org/abs/1901.07291 ）

- https://arxiv.org/abs/1901.02860
- https://ai.googleblog.com/2019/01/transformer-xl-unleashing-potential-of.html

**Comment by matchbou** [comment]
https://github.com/matchbou/Etc20190626
